service.test.js

```diff
@@ -1,18 +1,88 @@
 require('dotenv').config();
+const fs = require('fs');
 const axios = require('axios');
 const service = require('./service');
 
 jest.mock('axios');
 
+const readMockFile = (path) => fs.readFileSync(path, 'utf8');
+
 describe('Create Chapters', () => {
   afterEach(() => {
     jest.resetAllMocks();
   });
+
   const chapterApiURL = process.env.CONTENT_CHAPTER_API_URL;
   const apiKEY = process.env.API_KEY;
 
   test('creates one chapter sucessfully', async () => {
+    const rawMarkdownContent = readMockFile('__mocks__/fixtures/priv/markdown_templates/content/back-end/sql/_index.html.md');
+    const rawYamlContent = readMockFile('__mocks__/fixtures/priv/markdown_templates/content/back-end/sql/_index.yaml');
+
+    const returnData = {
+      data: {
+        chapter_id: 'c54f3049-965a-4634-ae16-6e4251ef7e3e',
+      },
+      status: 200,
+    };
+
+    const chapterBody = {
+      path: 'priv/markdown_templates/content/back-end/sql/table-management/_index.html.md',
+      markdownCommitId: 'd9771871c0cadc48e3d4141f93004b9c25d7201a',
+      contentMd: rawMarkdownContent,
+      yamlCommitId: 'd9771871c0cadc48e3d4141f93004b9c25d7201a',
+      contentYaml: rawYamlContent,
+    };
+
+    axios.post.mockReturnValue(Promise.resolve(returnData));
+    await expect(service.createVersion(chapterApiURL, chapterBody, apiKEY))
+      .resolves.toEqual(returnData);
+    expect(axios.post).toHaveBeenCalledTimes(1);
+  });
+
+  test('non hard-skills chapter couldn`t be created ', async () => {
+    const rawMarkdownContent = readMockFile('__mocks__/fixtures/priv/markdown_templates/content/soft-skills/creativity/_index.html.md');
+    const rawYamlContent = readMockFile('__mocks__/fixtures/priv/markdown_templates/content/soft-skills/creativity/_index.html.md');
+
+    const returnData = {
+      data: { message: 'Este conteúdo não é hard-skill e não foi salvo' },
+      status: 422,
+    };
 
+    const chapterBody = {
+      path: '__mocks__/fixtures/priv/markdown_templates/content/soft-skills/creativity/_index.html.md',
+      markdownCommitId: 'd9771871c0cadc48e3d4141f93004b9c25d7201a',
+      contentMd: rawMarkdownContent,
+      yamlCommitId: 'd9771871c0cadc48e3d4141f93004b9c25d7201a',
+      contentYaml: rawYamlContent,
+    };
+
+    axios.post.mockReturnValue(Promise.resolve(returnData));
+    await expect(service.createVersion(chapterApiURL, chapterBody, apiKEY))
```

Comment:
eu não entendi bem porque tem um teste que chama explicitamente o `service.createVersion` aqui. 

A responsabilidade de decidir se segue o processamento não é do `index.js`?

Reply:
na real eu acho que o título do que está confuso, né? 

como você está testando o service, ok chamar a criação de versão e ver se dá certo. 

mas o teste do fluxo fica pro do index.js. aí lá vc faz o cenário todo de ver se 400 impediu criar versão, se 422 fez criar normalmente.

Reply:
É um teste unitário mesmo da função do service de `createVersion`. 
Inclusive não pensei em uma forma de testar o `index.js` ainda se tiver alguma ideia, podem criar uma task pra isso ao final de toda integração

Reply:
@Caciquez  acho que é quase o que vc fez aqui. 

faz alguns cenários como: 

- se for 200, chama a criação de versão;
- se for 200 e 422, chama criação de versão;
- se tiver qualquer 400, assert que nao chamou a criação de versão;
- se tiver qualquer 500, assert que nao chamou a criação de versão;
- se for apenas 422, sem nenhum 200, nao criar versão 

faz sentido isso?

Reply:
Sim faz bastante sentido, vou criar um `index.test.js` e começar a testar esse fluxo por lá